packages/flow-cadut/src/imports.js

```diff
@@ -16,18 +16,9 @@
  * limitations under the License.
  */
 
-const getPairs = line => {
-  return line
-    .split(/\s/)
-    .map(item => item.replace(/\s/g, ""))
-    .filter(item => item.length > 0 && item !== "import" && item !== "from")
-}
-
-const collect = (acc, item) => {
-  const [contract, address] = item
-  acc[contract] = address
-  return acc
-}
+const REGEXP_IMPORT =
+  /(import\s*)((([\w\d]+)(\s*,\s*))*([\w\d]+))(\s+from\s*)([\w\d".\\/]+)/g
```

Comment:
Can you add some comments around this please? This is quite the regex. I know there are tests, but an explanation and examples would be really helpful for someone reading this. Might not even hurt to just test this alone as opposed to only testing a function it's used in.

Reply:
With groups bracket removed it can be simplified to this:
`/import\s+([\w\d]+)(\s*,\s*)*[\w\d]+\s+from\s+[\w\d".\\/]+/g`

Regexp explanation would be
- `/` - start of regexp
- `import\s*` - should have keyword `import` followed by any number of spaces. We need to fix it and replace `*` with `+` here cause we need at least single space...
- `([\w\d]+)(\s*,\s*)*([\w\d]+)\s+`
    -  `([\w\d]+)` - one or more word or digit - name of the contract
    - `(\s*,\s*)*` - zero or more comma character with zero or more leading and following space characters
    - `[\w\d]+` - one or more word or digit - name of another contract in multi import
- `\s+from\s+` - keyword `from` with one or more leading and following space characters
- `[\w\d".\\/]+` - one or more word, digit, `""` or `/` character for address or file import notation
- `/g` - end of regexp, `global` flag (find all)

Reply:
For a better DevEx we should actually use `cadence-parser`:
https://github.com/onflow/cadence/tree/master/npm-packages/cadence-parser

It was not production ready, when Cadut was in development 😅 
But maybe now it's time.

Reply:
The only problem with cadence-parser is that the WASM file is 4.6 MB.  If someone wanted to ship generated flow-cadut files to their frontend application, the overhead would be likely too much to justify.

Reply:
However - I can see some solution to this where the parser is used at the time of generation as oppose to at runtime, it just might require a non-trivial amount of work.